internal/artifacts/module_arch.go

```diff
@@ -108,117 +115,60 @@ func getBuildErrMsg(oneModuleMsg, manyModulesMsg string, modules []string) strin
 	return manyModulesMsg
 }
 
-func buildSelectedModules(source, target string, extensions []string, mtaObj *mta.MTA, selectedModules []string,
-	allDependencies bool, wdGetter func() (string, error)) error {
-
-	processedModules := make(map[string]bool)
+func collectSelectedModulesAndDependencies(mtaObj *mta.MTA, modulesWithDependencies map[string]bool, moduleName string) error {
 
-	for _, module := range selectedModules {
-		err := buildSelectedModule(source, target, extensions, mtaObj, module, allDependencies, processedModules, wdGetter)
+	modulesWithDependencies[moduleName] = true
```

Comment:
Add a check in the beginning that if `modulesWithDependencies` already contains this module (before setting it to `true`), don't do anything (to avoid going over the dependencies for the same module twice and also to avoid infinite loops in case of a loop in the dependencies)

Reply:
we have no loop in the dependencies, this is checked in buildModules function.

Reply:
Ok, but there's no need to calculate dependencies for modules you already calculated for.

Reply:
sure, as I wrote already, check added.